packages/versioning/src/versioning.ts

```diff
@@ -782,6 +781,48 @@ export enum Availability {
   Removed = "Removed",
 }
 
+function getParentAddedVersion(
+  program: Program,
+  type: Type,
+  versions: Version[]
+): Version | undefined {
+  let parentMap: Map<string, Availability> | undefined = undefined;
+  if (type.kind === "ModelProperty" && type.model !== undefined) {
+    parentMap = getAvailabilityMap(program, type.model);
+  } else if (type.kind === "Operation" && type.interface !== undefined) {
+    parentMap = getAvailabilityMap(program, type.interface);
+  }
+  if (parentMap === undefined) return undefined;
+  for (const [key, value] of parentMap.entries()) {
+    if (value === Availability.Added) {
+      return versions.find((x) => x.name === key);
+    }
+  }
+  return undefined;
+}
+
+function getParentAddedVersionInTimeline(
+  program: Program,
+  type: Type,
+  timeline: VersioningTimeline
+): Version | undefined {
+  let parentMap: Map<TimelineMoment, Availability> | undefined = undefined;
+  if (type.kind === "ModelProperty" && type.model !== undefined) {
+    parentMap = getAvailabilityMapInTimeline(program, type.model, timeline);
+  } else if (type.kind === "Operation" && type.interface !== undefined) {
+    parentMap = getAvailabilityMapInTimeline(program, type.interface, timeline);
+  }
+  if (parentMap === undefined) return undefined;
+  for (const [moment, availability] of parentMap.entries()) {
+    if (availability === Availability.Added) {
+      // FIXME: is this actually correct?
+      // We essentially want to return the first (earliest) time this is added.
+      return moment.versions().next().value;
```

Comment:
@timotheeguerin is this logic actually correct? I'm not as familiar with how the timeline is supposed to work.

Reply:
I don't think so all those version are the same moment, its just version for different libraries